sg/logger.go

```diff
@@ -1,56 +1,32 @@
 package sg
 
 import (
+	"context"
 	"fmt"
+	"log"
+	"os"
 	"strings"
 
-	"github.com/go-logr/logr"
-	"github.com/go-logr/logr/funcr"
-	"github.com/iancoleman/strcase"
+	"go.einride.tech/sage/internal/strcase"
 )
 
-// NewLogger returns a standard logger.
-func NewLogger(name string) logr.Logger {
-	return logr.New(&logSink{
-		name: strcase.ToKebab(strings.TrimPrefix(name, "main.")),
-		formatter: funcr.NewFormatter(funcr.Options{
-			RenderBuiltinsHook: func(kvList []interface{}) []interface{} {
-				// Don't render builtins.
-				return nil
-			},
-		}),
-	})
-}
-
-type logSink struct {
-	formatter funcr.Formatter
-	name      string
-}
-
-var _ logr.LogSink = &logSink{}
-
-func (s *logSink) Init(info logr.RuntimeInfo) {
-}
-
-func (s *logSink) Enabled(level int) bool {
-	return true
-}
+type loggerContextKey struct{}
 
-func (s *logSink) Info(level int, msg string, keysAndValues ...interface{}) {
-	_, args := s.formatter.FormatInfo(level, msg, keysAndValues)
-	fmt.Printf("[%s] %s%s\n", s.name, msg, args)
-}
-
-func (s *logSink) Error(err error, msg string, keysAndValues ...interface{}) {
-	_, args := s.formatter.FormatError(err, msg, keysAndValues)
-	fmt.Printf("[%s | ERROR] %s%s\n", s.name, msg, args)
+// NewLogger returns a standard logger.
+func NewLogger(name string) *log.Logger {
+	return log.New(os.Stderr, fmt.Sprintf("[%s] ", strcase.ToKebab(strings.TrimPrefix(name, "main."))), 0)
 }
 
-func (s *logSink) WithValues(keysAndValues ...interface{}) logr.LogSink {
-	panic("implement me")
+// WithLogger attaches a log.Logger to the provided context.
+func WithLogger(ctx context.Context, logger *log.Logger) context.Context {
+	return context.WithValue(ctx, loggerContextKey{}, logger)
 }
 
-func (s *logSink) WithName(name string) logr.LogSink {
-	s.name = name
-	return s
+// Logger returns the log.Logger attached to the provided context.
+func Logger(ctx context.Context) *log.Logger {
+	logger := ctx.Value(loggerContextKey{}).(*log.Logger)
+	if logger == nil {
+		panic("context is missing a Logger")
```

Comment:
Would it make sense to provide a default named logger if it's missing, instead of panicking?

Reply:
Hmm! Good point!

Reply:
Updated